content/courses/program-optimization/program-architecture.md

```diff
@@ -912,21 +923,56 @@ Now that our initial setup is ready, let's create our accounts. We'll have 3:
    - `game` - the game the monster is associated with
    - `hitpoints` - how many hit points the monster has left
 
+This is the final project structure:
+
+```bash
+src/
+├── constants.rs              # Constants used throughout the program
+├── error/                    # Error module
+│   ├── errors.rs             # Custom error definitions
+│   └── mod.rs                # Module declarations for error handling
+├── helpers.rs                # Helper functions used across the program
+├── instructions/             # Instruction handlers for different game actions
+│   ├── attack_monster.rs     # Handles attacking a monster
+│   ├── collect_points.rs     # Handles collecting points
+│   ├── create_game.rs        # Handles game creation
+│   ├── create_player.rs      # Handles player creation
+│   ├── mod.rs                # Module declarations for instructions
+│   └── spawn_monster.rs      # Handles spawning a new monster
+├── lib.rs                    # Main entry point for the program
+└── state/                    # State module for game data structures
+    ├── game.rs               # Game state representation
+    ├── mod.rs                # Module declarations for state
+    ├── monster.rs            # Monster state representation
+    └── player.rs             # Player state representation
+```
+
 When added to the program, the accounts should look like this:
 
 ```rust
 // ----------- ACCOUNTS ----------
-#[account]
-pub struct Game { // 8 bytes
-    pub game_master: Pubkey,            // 32 bytes
-    pub treasury: Pubkey,               // 32 bytes
-
-    pub action_points_collected: u64,   // 8 bytes
 
+// Inside `state/game.rs`
+use anchor_lang::prelude::*;
+#[account]
+#[derive(InitSpace)]
+pub struct Game {
+    pub game_master: Pubkey,
+    pub treasury: Pubkey,
+    pub action_points_collected: u64,
     pub game_config: GameConfig,
 }
 
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, InitSpace)]
+pub struct GameConfig {
+    pub max_items_per_player: u8,
+    pub for_future_use: [u64; 16], // Health of Enemies? Experience per item? Action Points per Action?
```

Comment:
Again double check re: account resizing. We might want to focus more on resizing and completely delete the concept of saving space for future.

Reply:
Yes removed these and added details on resizing using realloc with backward compatibility.